internal/cache/fifo.go

```diff
@@ -31,16 +32,23 @@ func NewFIFO[K comparable, V any](limit int) (*FIFO[K, V], error) {
 	return c, nil
 }
 
-func (f *FIFO[K, V]) Put(key K, val V) bool {
+// Put insert the [key, value] pair into the FIFO. In case an insert require the
+// removal of an existing key due to size limits, the evicted key would get returned.
+func (f *FIFO[K, V]) Put(key K, val V) (bool, *K) {
 	f.l.Lock()
 	defer f.l.Unlock()
 
+	var removedKey *K
 	_, exists := f.m[key]
 	if !exists {
+		f.removedKey = nil
 		f.buffer.Push(key) // Push removes the oldest [K] if we are at the [limit]
+		removedKey = f.removedKey
+		f.removedKey = nil
```

Comment:
Why do we add `f.removedKey = nil` both before and after? It seems we should only ever need one.